content.js

```diff
@@ -29,34 +47,58 @@ function toggleFiles() {
 }
 
 function addToggleBtn() {
-	const toggleBtn = createHtml('<a class="hide-files-btn btn btn-sm">Show dotfiles</a>');
+	const toggleBtn = createHtml(`<a class="hide-files-btn btn btn-sm">${label()}</a>`);
 	const btnContainer = document.querySelector('.file-navigation .right');
 
 	if (document.querySelector('.hide-files-btn')) {
+		addToggleBtnEvents();
 		return;
 	}
 
 	if (btnContainer) {
 		// insert after
 		btnContainer.insertBefore(toggleBtn, btnContainer.children[0]);
+		addToggleBtnEvents();
+	}
+}
 
-		const btn = document.querySelector('.hide-files-btn');
+function addToggleBtnEvents() {
+	const btn = document.querySelector('.hide-files-btn');
 
+	if (btn) {
 		btn.addEventListener('click', () => {
-			isHidden = !isHidden;
-			btn.textContent = isHidden ? 'Show dotfiles' : 'Hide dotfiles';
+			toggleOn = !toggleOn;
+			btn.textContent = label();
 			toggleFiles();
 		});
 	}
 }
 
+function label() {
+	return toggleOn ? 'Show dotfiles' : 'Hide dotfiles';
+}
+
 function trigger() {
 	addToggleBtn();
 	toggleFiles();
 }
 
 document.addEventListener('DOMContentLoaded', () => {
-	trigger();
-
 	new MutationObserver(trigger).observe(document.querySelector('#js-repo-pjax-container'), {childList: true});
+
+	window.chrome.storage.sync.get({
+		visibility: 'hidden',
+		ignoreRegEx: ''
```

Comment:
Or, if you think a "Restore Defaults" button is overkill, I can just use the default Regular Expression as the "Example" text that shows up below the input field. Letting the user copy and paste the defaults back into the Text Input.